default_adapter.go

```diff
@@ -10,55 +10,61 @@ import (
-	default_having_fragment         = []byte(" HAVING ")
-	default_order_by_fragment       = []byte(" ORDER BY ")
-	default_limit_fragment          = []byte(" LIMIT ")
-	default_offset_fragment         = []byte(" OFFSET ")
-	default_as_fragment             = []byte(" AS ")
-	default_asc_fragment            = []byte(" ASC")
-	default_desc_fragment           = []byte(" DESC")
-	default_nulls_first_fragment    = []byte(" NULLS FIRST")
-	default_nulls_last_fragment     = []byte(" NULLS LAST")
-	default_and_fragment            = []byte(" AND ")
-	default_or_fragment             = []byte(" OR ")
-	default_union_fragment          = []byte(" UNION ")
-	default_union_all_fragment      = []byte(" UNION ALL ")
-	default_intersect_fragment      = []byte(" INTERSECT ")
-	default_intersect_all_fragment  = []byte(" INTERSECT ALL ")
-	default_set_operator_rune       = '='
-	default_string_quote_rune       = '\''
-	default_place_holder_rune       = '?'
-	default_operator_lookup         = map[BooleanOperation][]byte{
+	replacement_rune                   = '?'
+	comma_rune                         = ','
+	space_rune                         = ' '
+	left_paren_rune                    = '('
+	right_paren_rune                   = ')'
+	star_rune                          = '*'
+	default_quote                      = '"'
+	period_rune                        = '.'
+	empty_string                       = ""
+	default_null                       = []byte("NULL")
+	default_true                       = []byte("TRUE")
+	default_false                      = []byte("FALSE")
+	default_update_clause              = []byte("UPDATE")
+	default_insert_clause              = []byte("INSERT INTO")
+	default_select_clause              = []byte("SELECT")
+	default_delete_clause              = []byte("DELETE")
+	default_truncate_clause            = []byte("TRUNCATE")
+	default_with_fragment              = []byte("WITH ")
+	default_recursive_fragment         = []byte("RECURSIVE ")
+	default_cascade_fragment           = []byte(" CASCADE")
+	default_retrict_fragment           = []byte(" RESTRICT")
+	default_default_values_fragment    = []byte(" DEFAULT VALUES")
+	default_values_fragment            = []byte(" VALUES ")
+	default_identity_fragment          = []byte(" IDENTITY")
+	default_set_fragment               = []byte(" SET ")
+	default_distinct_fragment          = []byte(" DISTINCT ")
+	default_returning_fragment         = []byte(" RETURNING ")
+	default_from_fragment              = []byte(" FROM")
+	default_where_fragment             = []byte(" WHERE ")
+	default_group_by_fragment          = []byte(" GROUP BY ")
+	default_having_fragment            = []byte(" HAVING ")
+	default_order_by_fragment          = []byte(" ORDER BY ")
+	default_limit_fragment             = []byte(" LIMIT ")
+	default_offset_fragment            = []byte(" OFFSET ")
+	default_for_update_fragment        = []byte(" FOR UPDATE ")
+	default_for_no_key_update_fragment = []byte(" FOR NO KEY UPDATE ")
```

Comment:
This doesn't seem universally supported.  Does a downstream user just override this if their rdbms uses different syntax?

Reply:
Also, it's an opt-in feature.  If you write a query using it and your DB doesn't support it, you'll just get an error when trying to run the query.  You'll then just have to rewrite your query to not use the feature that your DB doesn't support.